api/v1alpha2/cloudprovider_types.go

```diff
@@ -234,3 +242,26 @@ type CPILabelConfig struct {
 	// +optional
 	Region string `gcfg:"region,omitempty" json:"region,omitempty"`
 }
+
+// CPIRouteConfig is the vSphere cloud provider's route configuration.
+type CPIRouteConfig struct {
+	// T1RouterID is NSX-T T1 logical router ID used to create static route on Tier1 gateway.
+	// +optional
+	T1RouterID string `gcfg:"t1-router-id,omitempty" json:"t1-router-id,omitempty"`
+}
+
+// CPINSXTConfig is the vSphere cloud provider's NSX-T configuration.
+type CPINSXTConfig struct {
+	// SecretName is the name of the Kubernetes secret in which the NSX-T
+	// credentials are located.
+	// +optional
+	SecretName string `gcfg:"secret-name,omitempty" json:"secretName,omitempty"`
```

Comment:
Regarding secretName and secretNamespace, @dantingl will have a follow-up PR after https://github.com/kubernetes/cloud-provider-vsphere/pull/407 to support it

Reply:
Do you think we should support both secretName / secreNamespace AND raw fields for username/password? We do this in CPIGlobalConfig so maybe we should leave that option for users.

Reply:
gotcha, yeah I agree, will update!